src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1316,6 +1318,111 @@ testUnsafeSendChan TestTransport{..} = do
 
   takeMVar clientDone
 
+testRegistryMonitoring :: TestTransport -> Assertion
+testRegistryMonitoring TestTransport{..} = do
+  localNode <- newLocalNode testTransport initRemoteTable
+  remoteNode <- newLocalNode testTransport initRemoteTable
+  return ()
+
+  -- Local process. Test if local process will be removed from
+  -- registry then it dies.
+  box <- newEmptyMVar
+  runProcess localNode $ do
+    pid <- spawnLocal $ do
+                expect
+    register "test" pid
+    tpid <- whereis "test"
+    if tpid == Just pid
+      then do _ <- monitor pid
+              send pid ()
+              ProcessMonitorNotification{} <- expect
+              tpid1 <- whereis "test"
+              liftIO $ putMVar box (Nothing == tpid1)
+      else liftIO $ putMVar box False
+  True <- takeMVar box     
+  return ()
+
+  -- Remote process. Test if remote process entry is removed
+  -- from registry when process dies.
+  remote1 <- testRemote remoteNode 
+  runProcess localNode $
+    let waitpoll = do
+        w <- whereis "test" :: Process (Maybe ProcessId)
+        forM_ w (const waitpoll)
+    in do register "test" remote1
+          send remote1 ()
+          waitpoll
+          return ()
+  return ()
+    
+  -- Many labels. Test if all labels associated with process
+  -- are removed from registry when it dies.
+  remote2 <- testRemote remoteNode
+  runProcess localNode $
+    let waitpoll = do
+        w1 <- whereis "test-3" :: Process (Maybe ProcessId)
+        w2 <- whereis "test-4" :: Process (Maybe ProcessId)
+        forM_ (w1 <|> w2) (const waitpoll)
+    in do register "test-3" remote2
+          register "test-4" remote2
+          send remote2 ()
+          waitpoll
+          return ()
+
+{- XXX: waiting including patch for nsend for remote process
+  remote3 <- testRemote remoteNode
+  remote4 <- testRemote remoteNode
+  -- test many labels
+  runProcess localNode $ do
+     register "test-3" remote3
+     reregister "test-3" remote4
+     send remote3 ()
+     liftIO $ threadDelay 50000 -- XXX: racy
+     monitor remote4
+     nsend "test-3" ()
+     ProcessMonitorNotification{} <- expect
+     return ()
+-}
+
+  -- Test remodeAsync properties. Add a local process to
+  -- remote registry and checks if it will be removed
```

Comment:
-- remote registry and **check that it is** removed when **the** process **dies**.